### Simpson's Diversity Index Project

#### Introduction

Simpson's Diversity Index is a measure of diversity used in ecology. Here, we observed the number and type of insects in $1m^{2}$ x $1m^{2}$ plots in four different areas: a grassy area with frequent mowing, an area underneath concrete blocks, a landscaped area filled with mulch, and an area in an overwintering garden filled with leaves. We investigated the richness and evenness of types of insects, represented by the Simpson's Index of Diversity (1-D). The index runs from 0 to 1, with 1 being the most diverse and 0 being the least diverse. Usually, Simpson's Diversity Index is used to measure species evenness and richness; however, we only have lay knowledge of insects and did this project for fun. For this reason we grouped insects by lay terms as a replacement for "species."


#### Methods 

All measurements were taken within a 5-hour window on the same day: January 9th, 2023. The weather was slightly cloudy and the temperature was 48$^{\circ}$F (about 9$^{\circ}$C). 30 minutes of observation were conducted per plot. 

We investigated richness and evenness of types of insects, represented by the Simpson's Index of Diversity ($1-D$). Simpson's Index ($D$) measures the probability that two individuals randomly selected from a sample will belong to the same species (or some category other than species). The index runs from 0 to 1, with 1 being the most diverse and 0 being the least diverse. The equation to calculate $D$ used in this project is: 

$$ D = 	\Sigma (n / N ) ^{2} $$

where $n$ is the total number of a type of insect, and $N$ is the total number of all types of insects.

We then display measurements as $(1-D)$. 

#### Results: 

In [1]:
import pandas as pd
import numpy as np

df = pd.read_csv('BioIAData.csv')
df = df.rename(columns={'Unnamed: 0':'Insects'})
df.set_index('Insects')

Grass  Blocks  Mulch  Garden
Insects                                 
Beetle          0       1      0       0
Centipede       0       3      3      10
Earwig          0       0      1       0
Egg clump       0       2      0       0
Flea?           0       0      3       0
Fruit fly       1       1      1      14
Grass flea      1       2      0       0
Maggot          0       0      0       1
Milipede        0       1      1       3
Mosquito        0       0      0       1
Nematode        0       4      8       4
Pupa            0       1      0       0
Roly-Poly       0       2      0       0
Slug            0      11      0       5
Spider          0       8      0       2
Wood lice       0      14      7      14
Worm            2      15      0      18

In [128]:
sums = df.sum(axis=0)
print("The total number of organisms in the Grass area is:", sums['Grass'])
print("The total number of organisms in the Blocks area is:", sums['Blocks'])
print("The total number of organisms in the Mulch area is:", sums['Mulch'])
print("The total number of organisms in the Garden area is:", sums['Garden'])

The total number of organisms in the Grass area is: 4
The total number of organisms in the Blocks area is: 65
The total number of organisms in the Mulch area is: 24
The total number of organisms in the Garden area is: 72


In [125]:
richness = np.count_nonzero(df, axis=0)
Grass_Richness, Blocks_Richness, Mulch_Richness, Garden_Richness = richness[1], richness[2], richness[3], richness[4]
print("The richness of Grass area is:", Grass_Richness)
print("The richness of Blocks is:", Blocks_Richness)
print("The richness of Mulch is:", Mulch_Richness)
print("The richness of the Garden is:", Garden_Richness)

The richness of Grass area is: 3
The richness of Blocks is: 13
The richness of Mulch is: 7
The richness of the Garden is: 10


In [87]:
df.shape

(17, 5)

In [120]:
#Grass
grsum=0
for i in range(0,17):
    n = df.iloc[i,1]
    grsum = grsum + (n/sums['Grass'])**2
print("The Simpson's Index of Diversity (1-D) for Grass is:", round(1-grsum, 2))

#Blocks
bsum=0
for i in range(0,17):
    n=df.iloc[i,2]
    bsum = bsum + (n/sums['Blocks'])**2
print("The Simpson's Index of Diversity (1-D) for Blocks is:", round(1-bsum, 2))

#Mulch
msum=0
for i in range(0,17):
    n=df.iloc[i,3]
    msum = msum + (n/sums['Mulch'])**2
print("The Simpson's Index of Diversity (1-D) for Mulch is:", round(1-msum, 2))

#Mulch
gasum=0
for i in range(0,17):
    n=df.iloc[i,4]
    gasum = gasum + (n/sums['Garden'])**2
print("The Simpson's Index of Diversity (1-D) for Garden is:", round(1-gasum, 2))


The Simpson's Index of Diversity (1-D) for Grass is: 0.62
The Simpson's Index of Diversity (1-D) for Blocks is: 0.85
The Simpson's Index of Diversity (1-D) for Mulch is: 0.77
The Simpson's Index of Diversity (1-D) for Garden is: 0.83


#### Conclusion

The area with the highest diversity, as calculated with the Simpson's Diversity Index, was underneath the concrete blocks ($1-D$ = 0.85). The least diverse area was the grass lawn($1-D$ = 0.62). Generally, we observed that the more moisture and decaying organic matter was present, the greater the diversity of observed insects.